js/signal_protocol_store.js

```diff
@@ -136,15 +136,16 @@
         removePreKey: function(keyId) {
             var prekey = new PreKey({id: keyId});
 
-            new Promise(function(resolve) {
-                getAccountManager().refreshPreKeys().then(resolve);
-            });
-
-            return new Promise(function(resolve) {
-                prekey.destroy().then(function() {
-                    resolve();
-                });
-            });
+            return Promise.all([
+                new Promise(function(resolve) {
+                    getAccountManager().refreshPreKeys().then(resolve);
```

Comment:
My instinct says don't block on this, and if it fails, you still want to `removePreKey` to resolve. In fact, probably a better thing to do here is to decouple this class from the global getAccountManager by simply triggering a "removePreKey" event, and move the `refreshPreKeys` call to a listener in background.js. As a bonus, that would probably fix a lot of spurious console errors in the test page, and you could avoid the stubbing in storage_test.js.